distributed/scheduler.py

```diff
@@ -3431,8 +3419,8 @@ async def close_worker(self, worker: str, stimulus_id: str, safe: bool = False):
         """
         logger.info("Closing worker %s", worker)
         self.log_event(worker, {"action": "close-worker"})
-        # FIXME: This does not handle nannies
-        self.worker_send(worker, {"op": "close", "report": False})
+        ws = self.workers[worker]
+        self.worker_send(worker, {"op": "close", "nanny": bool(ws.nanny)})
```

Comment:
```suggestion
        self.worker_send(worker, {"op": "terminate"})
```
The worker can figure out what to do with the nanny on its own.

I'd like to use the new semantics and interfaces properly. `terminate` and `close` are similar-sounding words, but they now mean very different things. `terminate` means actually shut down. `close` means restart if there's a nanny, otherwise shut down.

I don't think the naming of these is clear enough, but what we want to do here, semantically, is `terminate`, so we should use that interface directly.